core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -376,7 +382,17 @@ private[spark] class MemoryStore(
 
     // Unroll this block safely, checking whether we have exceeded our threshold
     while (values.hasNext && keepUnrolling) {
-      serializationStream.writeObject(values.next())(classTag)
+      val value = values.next()
+      if (kryoSerializationStream != null) {
+        if (!kryoSerializationStream.classWrote) {
+          kryoSerializationStream.writeClass(value.getClass)
```

Comment:
@srowen you can see here. Here we don't use the writeAll, because we need acquire memory according to the written size.